primitives/src/vid/mod.rs

```diff
@@ -0,0 +1,77 @@
+//! Trait and implementation for a Verifiable Information Retrieval (VID).
+/// See <https://arxiv.org/abs/2111.12323> section 1.3--1.4 for intro to VID semantics.
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_std::{error::Error, fmt::Debug, string::String, vec::Vec};
+use displaydoc::Display;
+
+pub mod advz;
+
+/// The error type for `VidScheme` methods.
+#[derive(Display, Debug)]
+pub enum VidError {
+    /// invalid args: {0}
+    Argument(String),
+    /// internal error: {0}
+    Internal(anyhow::Error),
+}
+
+impl Error for VidError {}
+
+/// Convenience wrapper to convert any error into a [`VidError`].
+///
+/// Private fn so as not to expose error conversion API outside this crate
+/// as per [stackoverflow](https://stackoverflow.com/a/70057677).
+///
+/// # No-std support
+/// `no_std` mode requires `.map_err(vid)` to convert from a non-`anyhow` error
+/// as per [`anyhow` docs](https://docs.rs/anyhow/latest/anyhow/index.html#no-std-support),
+fn vid<E>(e: E) -> VidError
+where
+    E: ark_std::fmt::Display + Debug + Send + Sync + 'static,
+{
+    VidError::Internal(anyhow::anyhow!(e))
+}
+
```

Comment:
We could. I chose `vid` because you need to put `.map_err(vid)` everywhere, so I want a short name. I agree that `vid` on its own is not helpful, but `.map_err(vid)` seems clear to me.

Do you, as a fresh observer, prefer `vid_internal_err`? If so then I'm happy to change it.

Aside: sucks that this is state-of-the-art for error handling in Rust. 😕

Reply:
> because you need to put .map_err(vid) everywhere, so I want a short name.

yeah, I also guessed that's the reason. But yes, I do prefer clearer names here. Thanks for indulging me :)